types/datasource/proctree.go

```diff
@@ -0,0 +1,80 @@
+package datasource
+
+import "time"
+
+// ProcessInfo is the user facing representation of a process data at a specific time.
+type ProcessInfo struct {
+	Hash              uint32
+	Pid               int
+	NsPid             int
+	Ppid              int
+	ContainerId       string
+	Cmd               []string
+	ExecutionBinary   FileInfo
+	Interpreter       FileInfo
+	Interp            FileInfo
+	StartTime         time.Time // First thread fork time. TODO: Can we use the main thread start time instead?
+	ExecTime          time.Time // Last execve call time
+	ExitTime          time.Time
+	ParentHash        uint32
+	ThreadsIds        map[int]uint32 // Map between the ids to their hashes
+	ChildProcessesIds map[int]uint32 // Map between the ids to their hashes
+	IsAlive           bool
+}
+
+// ThreadInfo is the user facing representation of a thread data at a specific time.
+type ThreadInfo struct {
+	Hash        uint32
+	Tid         int
+	NsTid       int
+	Pid         int
+	UserId      int
+	GroupId     int
+	ForkTime    time.Time
+	ExitTime    time.Time
+	Name        string
+	ProcessHash uint32
+	IsAlive     bool
+}
+
+type FileInfo struct {
+	Path   string
+	Hash   string // TODO: should we call it SHA256 or Hash?
```

Comment:
lets call it Hash (and we can decide what algorithm to use, we can even change it later in newer versions without renaming the var).